metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/dao/ElasticsearchDao.java

```diff
@@ -100,7 +107,16 @@ public SearchResponse search(SearchRequest searchRequest) throws InvalidSearchEx
     searchResponse.setResults(Arrays.stream(elasticsearchResponse.getHits().getHits()).map(searchHit -> {
       SearchResult searchResult = new SearchResult();
       searchResult.setId(searchHit.getId());
-      searchResult.setSource(searchHit.getSource());
+      Map<String, Object> source;
+      if (fields.isPresent()) {
+        source = new HashMap<>();
+        searchHit.getFields().forEach((key, value) -> {
+          source.put(key, value.getValues().size() == 1 ? value.getValue() : value.getValues());
```

Comment:
If there are no results this line doesn't even get called because this code is inside a Stream map function that's iterating over the results.  The test case is really simple though so I added it just to be sure.